### ISSUE TITLE
indentation regression

### ISSUE THREAD

**Issue by dirk-thomas** [uncrustify/uncrustify#540]
The following code was "valid" and passed earlier `uncrustify` versions without any change. Commonly every open curly brace implies one additional level of indentation for the inside block. But the latest release as well as master insist on a weird indentation:

```
static struct Foo foo1[] = {
  {
  "first",
  2
}
};
```

The configuration file being used can be found [here](https://github.com/ament/ament_lint/blob/853968a9befe153e8c7d71659bbdceb18c97d9a2/ament_uncrustify/ament_uncrustify/configuration/ament_code_style.cfg).

A full example file with several comments what is expected and what is not:

```
struct Foo
{
  const char * first;
  int second;
};

// this block should not be changed
static struct Foo foo1[] = {
  {
    "first",  // uncrustify wants to remove two spaces from this line, starting in the same level as the open curly
    2  // same here
  } // this would even go to the beginning of the line
};

// this block should not be changed
static struct Foo foo2[] = {
  {"other",
   2},  // uncrustify wants to remove one space from this line
  {
  "last",  // uncrustify keeps this but that is not how I want it to look like
  2
  },  // uncrustify want to remove the indentation from this line too
};

// this block should actually be updated, adding indentation to the two inner lines
static struct Foo foo3[] = {
  {
  "last",  // uncrustify keeps this, but it should be indented by two space
  2  // same here
  },  // uncrustify want to remove the indentation from this line, which it should not
};
```

**Comment by dirk-thomas** [comment]
It seems to be related to `indent_continue`. In some cases it is not being honored.

**Comment by gmaurel** [comment]
Could you say which git version was still correct?

**Comment by dirk-thomas** [comment]
The last time I tried it with master was 85f466784214f6ca6c7883d010840753eb83a532 which still had the problem.

**Comment by gmaurel** [comment]
Well, even uncrustify 0.60 shows:
uncrustify 0.60
Parsing: dirk-thomas-1.cpp as language CPP
static struct Foo foo1[] = {
  {
    "first",
    2
  }
};
I don't anderstand what you mean.

**Comment by dirk-thomas** [comment]
Your last comment uses plain text so it is impossible to see what your output looks like.

Maybe it was a regression in an earlier version? Ubuntu only has version 0.59 for a long time (http://packages.ubuntu.com/trusty/devel/uncrustify). I am pretty sure the second block was valid in that version of uncrustify.

**Comment by dirk-thomas** [comment]
I just confirmed that the posted example code passes uncrustify 0.59 without any problem. So this is a regression.

**Comment by dirk-thomas** [comment]
I just confirmed that the posted example code is being formatted consistently with uncrustify 0.59. It correctly indents the lines within the curly braces. So this is a regression.

**Comment by mihaipopescu** [comment]
@dirk-thomas Do you think it could be related to issue #552 ? Can you try reverting that commit to see if that works for you ?

**Comment by dirk-thomas** [comment]
I had to resolve several conflicts when reverting the referenced commit but I can confirm that this fixed the regressions. With the commit reverted my code passes again as desired.

**Comment by mihaipopescu** [comment]
Yes, the revert should works as a temporary solution but we need something more solid than this. @gmaurel's approach to fix was good, i.e. to set the indentation to be the column of the next token after a brace open. That approach was already handled as well in the previous code when there was no newline between the brace and the next token. Also, the new code seems to be breaking the indentation of the brace close, alas the expected output of `30080-nl_brace_brace.cpp` doesn't look good at all.

The important thing is that the issue reproduces also with an `empty.cfg` and we can start from this.

**Comment by dirk-thomas** [comment]
As far as I can see the issue has been fixed with the following commit: 5c36767220a942f004b1b04b14181b1a231dc2c1
